src/mito2/src/compaction.rs

```diff
@@ -302,20 +279,88 @@ impl CompactionScheduler {
             for waiter in waiters {
                 waiter.send(Ok(0));
             }
-            return None;
+            self.region_status.remove(&region_id);
+            return Ok(());
+        };
+
+        // If specified to run compaction remotely, we schedule the compaction job remotely.
+        // It will fall back to local compaction if there is no remote job scheduler.
+        let waiters = if current_version.options.compaction.remote_compaction() {
+            if let Some(remote_job_scheduler) = &self.plugins.get::<RemoteJobSchedulerRef>() {
+                let remote_compaction_job = CompactionJob {
+                    compaction_region: compaction_region.clone(),
+                    picker_output: picker_output.clone(),
+                    start_time,
+                    waiters,
+                };
+
+                let result = remote_job_scheduler
+                    .schedule(
+                        RemoteJob::CompactionJob(remote_compaction_job),
+                        Box::new(DefaultNotifier {
+                            request_sender: request_sender.clone(),
+                        }),
+                    )
+                    .await;
+
+                match result {
+                    Ok(job_id) => {
+                        info!(
+                            "Scheduled remote compaction job {:?} for region {}",
```

Comment:
If we implement display for the job id, then we can use the display trait to print it
```suggestion
                            "Scheduled remote compaction job {} for region {}",
```